operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainStatus.java

```diff
@@ -587,10 +589,41 @@ public OffsetDateTime getInitialFailureTime() {
     return initialFailureTime;
   }
 
+  /**
+   * Returns the time that the last severe failure was reported.
+   */
   public OffsetDateTime getLastFailureTime() {
     return lastFailureTime;
   }
 
+  /**
+   * Return the number of times that failure timeouts should be increased as a result of failures.
+   * Will be approximately the number of times a failure has been reported, assuming one failure per retry.
+   * @param retrySeconds the number of seconds between retries.
+   */
+  public int getNumDeadlineIncreases(long retrySeconds) {
+    if (initialFailureTime == null || lastFailureTime == null) {
+      return 0;
+    } else {
+      return (int) (1 + divideRoundingUp(getSecondsFromInitialToLastFailure(), retrySeconds));
```

Comment:
The code seems to calculate how many times the failure has been retried based on the initialFailureTime, lastFailureTime and retry interval. But the retries do not happen back-to-back; each retry attempt takes sometime to fail especially here we are handling the situation where the job/pod timed out because the active deadline has exceeded.  The current calculation will yield a higher number than the actual number of times the retry happened.

Reply:
I think I see what you mean; since the intent is to come up with a reasonable approximation of the number of retries for the purpose of computing the new active deadline, we could presumably come up with a better algorithm that allows for the possibility that the failure was caused by an introspector timeout rather than one of the other failure modes. Can you think of a better algorithm that is not too complex? Ryan said that he wasn't worried about getting it precise, given that we are no longer tracking the actual number of failures.

Reply:
As Russ said here, I agree that the focus is on simplicity and transparency. I don't mind thinking about what the algorithm could be; however, if the operator is consistently communicating what the failure is and when the next retry will occur then I think this will be perfect for most customers -- put another way, very few if any customers will care that the next retry time is some exact value based on the number of retries if the time is communicated and is approximately what would be given by an algorithm that did have the exact number of retires.